src/create_shards.c

```diff
@@ -389,26 +394,38 @@ ParseWorkerNodeFile(char *workerNodeFilename)
 	}
 
 	/* build pattern to contain node name length limit */
-	snprintf(workerLinePattern, sizeof(workerLinePattern), "%%%us%%*[ \t]%%10u",
-			 MAX_NODE_LENGTH);
+	snprintf(workerLinePattern, sizeof(workerLinePattern), workerPatternTemplate,
+			 MAX_NODE_LENGTH, MAX_PORT_LENGTH);
 
 	while (fgets(workerNodeLine, sizeof(workerNodeLine), workerFileStream) != NULL)
 	{
+		const int workerLineLength = strnlen(workerNodeLine, MAXPGPATH);
 		WorkerNode *workerNode = NULL;
 		char *linePointer = NULL;
-		uint32 nodePort = 0;
-		int parsedValues = 0;
+		int32 nodePort = PostPortNumber; /* default port number */
+		int fieldCount = 0;
+		bool lineIsInvalid = false;
 		char nodeName[MAX_NODE_LENGTH + 1];
+		char nodePortString[MAX_PORT_LENGTH + 1];
 		memset(nodeName, '\0', sizeof(nodeName));
+		memset(nodePortString, '\0', sizeof(nodePortString));
 
-		if (strnlen(workerNodeLine, MAXPGPATH) == MAXPGPATH - 1)
+		if (workerLineLength == MAXPGPATH - 1)
 		{
 			ereport(ERROR, (errcode(ERRCODE_CONFIG_FILE_ERROR),
 							errmsg("worker node list file line exceeds the maximum "
 								   "length of %d", MAXPGPATH)));
 		}
 
-		/* skip leading whitespace and check for # comment */
+		/* trim trailing newlines preserved by fgets, if any */
+		linePointer = workerNodeLine + workerLineLength - 1;
+		while (linePointer >= workerNodeLine &&
```

Comment:
@jasonmp85 Can we get rid of this while loop if we update the `workerPatternTemplate` as follows (ie: add `\n\r` for host and port patterns?):
```C
char *workerPatternTemplate = "%%%u[^# \t\n\r]%%*[ \t]%%%u[^# \t\n\r]";
```
Does that make sense? Or, should we leave as it is?